presto-main/src/main/java/com/facebook/presto/operator/aggregation/PrecisionRecallAggregation.java

```diff
@@ -0,0 +1,192 @@
+            @SqlType(StandardTypes.BIGINT) long bucketCount,
+            @SqlType(StandardTypes.BOOLEAN) boolean outcome,
+            @SqlType(StandardTypes.DOUBLE) double pred,
+            @SqlType(StandardTypes.DOUBLE) double weight)
+    {
+        if (state.getTrueWeights() == null) {
+            state.setTrueWeights(new FixedDoubleHistogram(
+                    (int) (bucketCount),
+                    PrecisionRecallAggregation.MIN_PREDICTION_VALUE,
+                    PrecisionRecallAggregation.MAX_PREDICTION_VALUE));
+            state.setFalseWeights(new FixedDoubleHistogram(
+                    (int) (bucketCount),
+                    PrecisionRecallAggregation.MIN_PREDICTION_VALUE,
+                    PrecisionRecallAggregation.MAX_PREDICTION_VALUE));
+        }
+
+        if (pred < MIN_PREDICTION_VALUE || pred > PrecisionRecallAggregation.MAX_PREDICTION_VALUE) {
+            throw new PrestoException(
+                    INVALID_FUNCTION_ARGUMENT,
+                    PrecisionRecallAggregation.ILLEGAL_PREDICTION_VALUE_MESSAGE);
+        }
+        if (weight < 0) {
+            throw new PrestoException(
+                    INVALID_FUNCTION_ARGUMENT,
+                    PrecisionRecallAggregation.NEGATIVE_WEIGHT_MESSAGE);
+        }
+        if (bucketCount != state.getTrueWeights().getBucketCount()) {
+            throw new PrestoException(
+                    INVALID_FUNCTION_ARGUMENT,
+                    PrecisionRecallAggregation.INCONSISTENT_BUCKET_COUNT_MESSAGE);
+        }
+
+        if (outcome) {
+            state.getTrueWeights().add(pred, weight);
+        }
+        else {
+            state.getFalseWeights().add(pred, weight);
+        }
+    }
+
+    @InputFunction
+    public static void input(
+            @AggregationState PrecisionRecallState state,
+            @SqlType(StandardTypes.BIGINT) long bucketCount,
+            @SqlType(StandardTypes.BOOLEAN) boolean outcome,
+            @SqlType(StandardTypes.DOUBLE) double pred)
+    {
+        PrecisionRecallAggregation.input(state, bucketCount, outcome, pred, PrecisionRecallAggregation.DEFAULT_WEIGHT);
+    }
+
+    @CombineFunction
+    public static void combine(
+            @AggregationState PrecisionRecallState state,
+            @AggregationState PrecisionRecallState otherState)
+    {
+        if (state.getTrueWeights() == null && otherState.getTrueWeights() != null) {
+            state.setTrueWeights(new FixedDoubleHistogram(otherState.getTrueWeights()));
+            state.setFalseWeights(new FixedDoubleHistogram(otherState.getFalseWeights()));
+            return;
+        }
+        if (state.getTrueWeights() != null && otherState.getTrueWeights() != null) {
+            state.getTrueWeights().mergeWith(otherState.getTrueWeights());
+            state.getFalseWeights().mergeWith(otherState.getFalseWeights());
+        }
+    }
+
+    protected static class BucketResult
+    {
+        public final Double totalTrueWeight;
```

Comment:
Just a reminder on the use of `Double`.  The reason I'm saying this is two fold:

1) It incurs object allocation overhead that didn't otherwise exist 
2) It makes one read this code wonder if there's a null case, which isn't possible for primitive double

From what I've read, I don't see any case where any of these could be null.